src/Microsoft.TestPlatform.Extensions.TrxLogger/XML/XmlPersistence.cs

```diff
@@ -689,6 +693,15 @@ public XmlNode EnsureLocationExists(XmlElement xml, string location)
             return this.EnsureLocationExists(xml, location, this.namespaceUri);
         }
 
+        private string RemoveInvalidXmlChar(string str)
+        {
+            // From xml spec (http://www.w3.org/TR/xml/#charsets) valid chars: 
+            // #x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF]  
+
+            string invalidChar = @"[^\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFD\u10000-\u10FFFF]";
+            return Regex.Replace(str, invalidChar, TrxLoggerResources.PlaceHolderForInvalidChar);
```

Comment:
I am not sure if `?` should be localized. Secondly, questionmark may have a different representation in various languages, are those values in the *allowed characters in xml* bucket?

Probably better to leave `?` as it is. It is an indicator for missing data.